p2p/providers/peerstore_provider/peerstore_provider.go

```diff
@@ -29,7 +29,10 @@ func ActorsToPeerstore(abp PeerstoreProvider, actors []*coreTypes.Actor) (pstore
 	pstore = make(sharedP2P.PeerAddrMap)
 	for _, a := range actors {
 		networkPeer, err := ActorToPeer(abp, a)
-		if err != nil {
+		if _, ok := err.(*ErrResolvingAddr); ok {
+			logger.Global.Warn().Err(err).Msg("ignoring ErrResolvingAddr - peer unreachable, not adding it to peerstore")
+			continue
+		} else if err != nil {
```

Comment:
@Olshansk @deblasis wdyt about a `// TECHDEBT(#519)` comment here. This looks to me like a place where we may be better served in the long run by asserting for behavior rather than type (i.e. asserting against an interface type -  see Dave Cheney clip):

- ![image](https://user-images.githubusercontent.com/600733/227512535-fdf3f63a-a64f-429d-a60e-ef9d44ed0325.png)
- “Don’t just check errors, handle them gracefully”  — Dave Cheney (2016): [https://www.youtube.com/watch?v=lsBF58Q-DnY&list=WL&index=7](https://www.youtube.com/watch?v=lsBF58Q-DnY&list=WL&index=7)
- Other example of sentinel errors being problematic  (`context` context): stdlib contexts: https://share.snipd.com/chapter/7620f9f8-0a3d-41fa-8118-01663a51cb40

Reply:
Thanks @bryanchriswhite! https://github.com/pokt-network/pocket/pull/611/commits/0f40d94f37ccbc8fe52423fa815b0e4b42538464